plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/NullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,178 @@
+    {
+        checkState(!testCases.isEmpty(), "No test cases");
+        try (TemporaryRelation temporaryRelation = dataSetup.setupTemporaryRelation(unmodifiableList(testCases))) {
+            verifyPredicate(queryRunner, session, temporaryRelation, true, false, !connectorExpressionOnly);
+            verifyPredicate(queryRunner, session, temporaryRelation, true, true, true);
+            verifyPredicate(queryRunner, session, temporaryRelation, false, false, !connectorExpressionOnly);
+            verifyPredicate(queryRunner, session, temporaryRelation, false, true, true);
+        }
+        return this;
+    }
+
+    private void verifyPredicate(QueryRunner queryRunner, Session session, TemporaryRelation temporaryRelation, boolean isNull, boolean connectorExpression, boolean expectPushdown)
+    {
+        TestCase firstCase = testCases.getFirst();
+        String firstColumnName = "col_0";
+        String withConnectorExpression = connectorExpression ? " OR %s IS NULL".formatted(firstColumnName) : "";
+
+        String queryWithAll = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " +
+                IntStream.range(0, testCases.size())
+                        .mapToObj(column -> getPredicate(column, isNull))
+                        .flatMap(Optional::stream)
+                        .collect(joining(" AND "))
+                + withConnectorExpression;
+
+        // Closing QueryAssertions would close the QueryRunner
+        QueryAssertions queryAssertions = new QueryAssertions(queryRunner);
+        try {
+            assertPushdown(expectPushdown,
+                    assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                            assertThat(queryAssertions.query(session, queryWithAll))));
+        }
+        catch (AssertionError e) {
+            // if failed - identify exact column which caused the failure
+            for (int column = 0; column < testCases.size(); column++) {
+                String queryWithSingleColumnPredicate = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " + getPredicate(column, isNull) + withConnectorExpression;
+                assertPushdown(expectPushdown,
+                        assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                                assertThat(queryAssertions.query(session, queryWithSingleColumnPredicate))));
+            }
+            throw new IllegalStateException("Single column assertion should fail for at least one column, if query of all column failed", e);
+        }
+    }
+
+    private Optional<String> getPredicate(int column, boolean isNull)
+    {
+        if (specialColumns.contains(column)) {
+            // Special columns should not be a part of a predicate, as they are helpers and do not participate in the assertions
+            return Optional.empty();
```

Comment:
Instead can we return `TRUE` as an expression which would be used removed by the optimizer